models/tenant.py

```diff
@@ -16,17 +17,24 @@ class TenantModel(BaseModel):
     # relationships
     staff = relationship("UserModel", secondary="staff_tenant_links")
     leases = db.relationship(
-        "LeaseModel", backref="tenant", lazy=True, cascade="all, delete-orphan"
+        "LeaseModel",
+        backref="tenant",
+        # lazy="dynamic" is required so that we can filter() over leases in
+        # json(). See: https://stackoverflow.com/questions/11578070
+        lazy="dynamic",
```

Comment:
I tend to not like comments in code. I think this comment would be good to put in the commit that added this, then if anyone has any question as to what this does they can either look it up in the documentation or review the commit history and see what the contributor wrote in the commit log as to why this was added.

Reply:
Done; moved to commit message.